GCP: Refactor and update docs

Edwin-Pau opened the pull request:
Refactored encryption script's read_tfvars function into one and updated the documentation in regards to the encryption script.

Commit:
GCP: Update documentation for encryption script

Updated documentation to make it more clear and less wordy.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
GCP: Refactor encryption script read_tfvars

Refactored the two methods read_tfvars_data and read_tfvars_secrets
into one function read_tfvars since they share so much in common.

Signed-off-by: Edwin-Pau <epau@teradici.com>

shermanyin commented on tools/kms_secrets_encryption.py:
```diff
@@ -445,8 +415,7 @@ def read_tfvars_secrets(self, tfvars_file):
                     begin_reading_secrets = True
-                    continue
-                
+
```
continue here?

shermanyin commented on tools/kms_secrets_encryption.py:
```diff
@@ -456,8 +425,12 @@ def read_tfvars_secrets(self, tfvars_file):
                     tf_secrets[key] = value.replace("\"", "")
+                else:           
+                    # Split the line into key value pairs using the first delimiter
+                    key, value = map(str.strip, line.split('=', 1))
```
this line should happen before the if, don't need to be repeated.

shermanyin commented on docs/README-gcp.md:
````diff
@@ -50,14 +50,16 @@ To encrypt secrets using the KMS crypto key created above, follow the instructio
 ```./kms_secrets_encryption.py </path/to/terraform.tfvars>```
 
-The script rewrites your existing terraform.tfvars secrets using ciphertext so that it is ready to be used for your Terraform deployment. In addition, any text files specified under the secrets as a path will also be encrypted so that it is ready to be used by Terraform. The script will also update the path to the encrypted text file inside of the new terraform.tfvars.
+The script will replace all your plaintext secrets inside of terraform.tfvars with ciphertext. Any text files specified under the secrets section as a path will also be encrypted. 
````
avoid using 2nd person.  "will replace all the plaintext"...

shermanyin commented on docs/README-gcp.md:
```diff
@@ -50,14 +50,16 @@ To encrypt secrets using the KMS crypto key created above, follow the instructio
+Alternatively, the kms_secrets_encryption.py Python 3 script under the tools directory can be used to automate the KMS encryption process. 
 
-Execute the following command inside the tools directory:
+First, open the terraform.tfvars file and enter all the secrets in plaintext located under the line "# <-- Start of secrets section, do not edit this line. -->", then save the file. 
```
can you use number list?
1. step one
2. step two, etc.

IMO it's easier to read. then you don't have to say "First,"

shermanyin commented on tools/README.md:
```diff
@@ -17,8 +17,9 @@ This is a Python 3 script that automates the encryption and decryption of secret
-ciphertext in it instead of the plaintext and will append ".encrypted" to the end of the file name. Finally, the script will
+
+If the secret is a path to a text file, the text file itself will be encrypted instead. The script will encrypt the
+plaintext in the textfile and will also append ".encrypted" to the end of the file name. Finally, the script will
```
"will encrypt the file" is good enough

Commit:
GCP: Refactor encryption script

Refactored the reading tfvars function when reading key value pairs
from the tfvars file. The line to read key value pairs has been moved
to outside the if block. A continue statement has been added back
after detecting the start of the secrets flag.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
GCP: Update encryption script docs

Updated encryption script section in GCP docs and in tools docs.

Signed-off-by: Edwin-Pau <epau@teradici.com>
